### PR TITLE
feat: create base for CodeBlockPlugin

### PR DESCRIPTION
**Pull request by TomasSlama**
[FX-4184]

### Description

This PR brings the default structure for the `CodeBlockPlugin`
- stories
- toolbar button
- RichText component
- styles
- types

The logic will be in different PR

### How to test

- [Temploy](https://picasso.toptal.net/fx-4184-rte-code-block)
- check the code and [the story](https://picasso.toptal.net/fx-4184-rte-code-block/?path=/story/forms-richtexteditor--richtexteditor#code)

### Screenshots

image

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests

**Breaking change**

- codemod is created and showcased in the changeset
- test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4184]: https://toptal-core.atlassian.net/browse/FX-4184?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/picasso-rich-text-editor/src/LexicalEditor/styles.ts [hunk 1]
```diff
@@ -165,5 +165,12 @@ export default (theme: Theme) => {
       },
     },
     code: codeStyles(theme),
+    codeBlock: codeBlockStyles(theme),
+    codeBlockText: {
+      fontFamily: 'inherit',
+      fontSize: 'inherit',
+      lineHeight: 'inherit',
+      color: 'inherit',
+    },
```
Comment (TomasSlama) [inline packages/picasso-rich-text-editor/src/LexicalEditor/styles.ts:174]: Lexical has all text nodes wrapped in span, but RichText doesn't have such spans. To reuse all styles between RTE and RT I decided to inherit the styles from the wrapper.

File: packages/picasso-rich-text-editor/src/plugins/CodeBlockPlugin/CodeBlockButton.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import RichTextEditorButton from '../../RichTextEditorButton'
+
+export type Props = {
+  'data-testid'?: string
+}
+
+const CodeBlockButton = ({ 'data-testid': testId }: Props) => {
+  const { disabled, focused } = useRTEPluginContext()
+
+  const handleClick = () => {}
```
Comment (TomasSlama) [inline packages/picasso-rich-text-editor/src/plugins/CodeBlockPlugin/CodeBlockButton.tsx:14]: will be specified in separate PR

**Comment by TomasSlama** [comment]
@toptal-anvil ping reviewers

**Comment by TomasSlama** [comment]
@toptal-anvil ping reviewers

File: packages/picasso-rich-text-editor/src/utils/html-to-hast.ts [hunk 3]
```diff
@@ -13,6 +13,7 @@ export const hastSanitizeSchema: Schema = {
   attributes: {
     a: ['href'],
     img: ['src', 'data*', 'class'],
+    pre: ['data*', 'spellcheck'],
```
Comment (mkrl) [inline packages/picasso-rich-text-editor/src/utils/html-to-hast.ts:16]: Is having `spellcheck` a requirement? I wonder if it will actually be stored in HAST.

**Review by mkrl — APPROVED** [review]
Code looks good, the UI seems accurate. My only concern is grouping Code and inline code buttons together in the toolbar (this is a common problem across all the plugins, they are all separated while some must be grouped.

File: packages/picasso-rich-text-editor/src/utils/html-to-hast.ts (hunk 3 above)
Comment (TomasSlama) [reply packages/picasso-rich-text-editor/src/utils/html-to-hast.ts:16]: good point, I will remove it
